File: app/src/main/java/com/bkjcb/rqapplication/base/util/FileUtil.java
package com.bkjcb.rqapplication.base.util;

import com.bkjcb.rqapplication.base.MyApplication;

import java.io.File;

/**
 * Created by DengShuai on 2020/6/16.
 * Description :
 */
public class FileUtil {
    public static String getFileOutputPath(String s) {
        File file = new File(MyApplication.getContext().getExternalFilesDir(null).getAbsolutePath(), s);
        if (!file.exists() && file.mkdirs()) {
            return file.getAbsolutePath();
        }
        return MyApplication.getContext().getFilesDir().getAbsolutePath();
    }
}

File: app/src/main/java/com/bkjcb/rqapplication/base/util/FileUtil.kt
package com.bkjcb.rqapplication.base.util

import com.bkjcb.rqapplication.MyApplication
import java.io.File

/**
 * Created by DengShuai on 2020/6/16.
 * Description :
 */
object FileUtil {
    fun getFileOutputPath(s: String?): String {
        val file = File(MyApplication.getContext().getExternalFilesDir(null)?.absolutePath, s)
        return if (!file.exists() && file.mkdirs()) {
            file.absolutePath
        } else MyApplication.getContext().filesDir.absolutePath
    }
}